src/LPOracle.sol

```diff
@@ -182,9 +182,6 @@ contract LPOracle is AggregatorV3Interface {
             balance0 -= order.sellAmount;
             balance1 += order.buyAmount;
         }
-
-        /* Adjust for decimals */
-        (token0Bal, token1Bal) = _adjustDecimals(balance0, balance1, TOKEN0_DECIMALS, TOKEN1_DECIMALS);
     }
```

Comment:
So this was the source of the bug? Using adjust decimals in a place we weren't meant to (I recall thinking it was weird we passed in the fixed token decimals here and not in the other place). adjustDecimals is for feed data:

https://github.com/cowprotocol/cow-amm/blob/82229b9d7e18abe1ebe1d1ee1d833821e48c31bf/src/oracles/ChainlinkPriceOracle.sol#L73

Reply:
Yeah, this correction is a better approach since we eliminate this specific bug & we remain less opinionated about pool token decimal requirements. Previously we would have had to enforce underlying pool tokens <= 18 if going via _adjustDecimals.

This new approach:
- adjusts the balance * price values by their corresponding decimals and ensures the output has base decimals of 18. 
- assumes the LP token ERC-20 implementations use 18 decimals, which they do.